_posts/2018-01-02-getting-up-and-running-with-react-native.md

```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title: Getting up and running with React Native
+date: 2018-01-02
+---
+
+Today, I meant to look into [react-native](https://facebook.github.io/react-native/). I know React (a little) but recently realised I had made some wrong assumptions about React Native: I thought the idea behind React and React-native was: **Learn once, write once, use everywhere**. It is NOT! The real motto is: **Learn once, use everywhere**, i.e. you learn the basics once but you can't have one code base targeting all platforms. What a disappointment... I thought with the help of some kind of web dark magic it was the case.
+
+Anyway, since it is not the case and one needs to learn a few things that are React Native specific, off I went to learn the basic of React Native. It actually took me quite a while to get up and running as I met quite a few hick ups along the way.
+<br/>
+I planned to go through this [tutorial](https://facebook.github.io/react-native/docs/getting-started.html) on React Native website. In this post I am sharing my experience with the Getting Started section (yep, my post is actually longer than the Getting Started page haha).
+
+#### Step 1: install the tool create-react-native-app
+```
+npm install -g create-react-native-app
+```
+
+=> ✔️ Successful!
+
+#### Step 2: create a React Native app project
+```
+create-react-native-app HelloWorld
+```
+
+=> ❌ Unfortunately, failed! It didn't do much, just created a package.json file and then failed. The good news is that it failed with a very useful error message.
+
+![create-react-native-app error]({{ "/assets/blog/create-react-native-app-error.png" | absolute_url }}){:height="90%" width="90%"}
+
+I googled a bit trying to understand what was going on and why npm 5 wouldn't be supported by [create-react-native-app](https://github.com/react-community/create-react-native-app#installation). I haven't found the exact answer yet apart from `npm 5 is buggy`. The main suggestion to fix this problem is to downgrade npm to 4.x.x, but I don't really like the idea of downgrading, I think we should move forward, not backwards. So I searched a bit more and luckily came across a very useful comment on this [issue](https://github.com/react-community/create-react-native-app/issues/424):
```

Comment:
_isn't_ ~wouldn't be~ supported by

Reply:
This is a really interesting insight, don't know anything about this but not being able to move up a major version seems like a pretty big deal!

Reply:
It is so frustrating not to know why. I can't find the answer :(